flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/autoscaler/KubernetesScalingRealizerTest.java

```diff
@@ -30,12 +31,23 @@
 public class KubernetesScalingRealizerTest {
 
     @Test
-    public void testAutoscalerOverridesVertexIdsAreSorted() {
-
+    public void testAutoscalerOverridesStringDoesNotChangeUnlessOverridesChange() {
         KubernetesJobAutoScalerContext ctx =
                 TestingKubernetesAutoscalerUtils.createContext("test", null);
+        FlinkDeployment resource = (FlinkDeployment) ctx.getResource();
+
+        // Create resource with existing parallelism overrides
+        resource.getSpec()
+                .getFlinkConfiguration()
+                .put(PipelineOptions.PARALLELISM_OVERRIDES.key(), "a:1,b:2");
```

Comment:
Would it make sense to reverse this to `"b:2,a1" ` so that the test actually fails without the change?

Reply:
It's HashMap instead of TreeMap or SortedMap, so the ordering cannot be determined, right?

If so, I prefer to test both of `"a:1,b:2"` and `"b:2,a:1"`, wdyt?